ext/adaptors/ranges-lib.tex

```diff
@@ -3186,46 +3208,72 @@
 
 \indexlibrary{\idxcode{begin}!\idxcode{join_view}}%
 \begin{itemdecl}
-constexpr iterator begin();
-constexpr const_iterator begin() const requires InputRange<const R> &&
+constexpr @\changed{iterator}{auto}@ begin();
+constexpr @\changed{const_iterator}{auto}@ begin() const requires InputRange<const R> &&
   is_reference_v<reference_t<iterator_t<const R>>>;
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
-\returns \tcode{\{*this, ranges::begin(base_)\}}.
+\effects \added{Equivalent to:}
+\begin{codeblock}
+@\added{return \xname{iterator}<\textit{simple-view}<R{>}>}@{*this, ranges::begin(base_)}@\added{;}@
+\end{codeblock}
+\begin{addedblock}
+and
+\begin{codeblock}
+return @\xname{iterator}@<true>{*this, ranges::begin(base_)};
+\end{codeblock}
+for the first and second overloads, respectively.
+\end{addedblock}.
 \end{itemdescr}
 
 \rSec4[ranges.adaptors.join_view.end]{\tcode{join_view} range end}
 
 \indexlibrary{\idxcode{end}!\idxcode{join_view}}%
 \begin{itemdecl}
-constexpr sentinel end();
-constexpr const_sentinel end() const requires InputRange<const R> &&
+constexpr @\changed{sentinel}{auto}@ end();
+constexpr @\changed{const_sentinel}{auto}@ end() const requires InputRange<const R> &&
   is_reference_v<reference_t<iterator_t<const R>>>;
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
-\effects Equivalent to \tcode{sentinel\{*this\}} and
-\tcode{const_sentinel\{*this\}} for the first and second overload,
-respectively.
+\effects Equivalent to\added{:}
+\begin{codeblock}
+@\added{return} \changed{sentinel}{\xname{sentinel}<\textit{simple-view}<R>{>}}\{*this\}\added{;}@
+\end{codeblock}
+  and
+\begin{codeblock}
+@\added{return} \changed{const_sentinel}{\xname{sentinel}<true>}\{*this\}\added{;}@
+\end{codeblock}
+for the first and second overload, respectively.
```

Comment:
I'm wary of having two members that do the same thing in the `simple-view<R>` case; it increases code bloat if the linker doesn't perform identical comdat folding, and just causes confusion in general. I would prefer that the non-`const` overload requires `!simple-view<R>` and just disappears. (This is admittedly a persnickety detail this early in the design process.)